core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -121,6 +121,17 @@ private ManifestFiles() {
     throw new UnsupportedOperationException("Cannot write manifest for table version: " + formatVersion);
   }
 
+  /**
+   * Returns a new {@link ManifestReader} for a {@link ManifestFile}.
+   *
+   * @param manifest a {@link ManifestFile}
+   * @param io a {@link FileIO}
+   * @return a {@link ManifestReader}
+   */
+  public static ManifestReader<DeleteFile> readDeleteManifest(ManifestFile manifest, FileIO io) {
```

Comment:
Do we plan to need this in future?

Reply:
I think so ,  if people don't provide a `specsById`, then it good to parse the spec from table's metadata and get the partition specification.  The  FlinkManifestUtil#readCompletedFiles is deserializing the DeleteFiles from delete manifest files by using the `readDeleteManifest(manifest, io, null);`